trunner/tools/gdb.py

```diff
@@ -158,10 +166,28 @@ class OpenocdGdbServer(OpenocdProcess):
     def run(self):
         try:
             self._start()
-            yield
+            yield self
         finally:
             self._close()
 
+    def _close(self):
+        if not self.proc:
+            return
+
+        clear_pexpect_buffer(self.proc)
+        self.proc.close(force=True)
+
+        # force=True terminates the process, results in exitstatus=None and signalstatus=SIGHUP
+        # this is expected behavior in this case, as OpenOCD is running as a server and we want to kill it
+        if self.proc.exitstatus not in (0, None):
+            raise OpenocdError(f"OpenOCD exited with status {self.proc.exitstatus}", output=self.host_log.getvalue())
+
+        if self.proc.signalstatus != 1:
```

Comment:
you're using implementation detail of pexpect and openocd (start killing with `SIGHUP`, assuming openocd will die on `SIGHUP`). 

IMHO we should call `_close(timeout=1)` from the base class but treat the result in a special way (swallow the exception if it was expected).

Probably the best way to finish the `openocd` gracefully would be to send `exit` command to it and wait for process finish (no need to do it right away, but maybe worth adding a TODO)